src/menu.js

```diff
@@ -0,0 +1,21 @@
+//f. updating the dictionary and generating the hyperlinks
+function refresh() {
+  const ss = SpreadsheetApp.openByUrl(url);
+  const metadataAA = ss.getSheetByName("metadata").getRange(2,1,ss.getSheetByName("metadata").getLastRow(),1).getValues();
+  ss.getSheetByName("A Fronte").getRange(2,1,ss.getSheetByName("A Fronte").getLastRow(),3).clear();
+  ss.getSheetByName("A Fronte").getRange(2,1,ss.getSheetByName("metadata").getLastRow(),1).setValues(metadataAA);
+  ss.getSheetByName("A Fronte").getRange(2,2,ss.getSheetByName("A Fronte").getLastRow()-1,1).insertCheckboxes();
+  generate_atergo ();
+  links_in_ss ('A Fronte','A Tergo');
+}
+
+//f. creating a menu with refresh option
+function addMenu() {
+  var menu = SpreadsheetApp.getUi().createMenu('Dodatkowe');
+  menu.addItem('Odśwież', 'refresh');
+  menu.addToUi();
+}
+
+function onOpen(e) {
```

Comment:
Not sure why, but onOpen didnt add an element to the menu by default - I needed to add it as a rule to call onOpen when sheet is opened... Am I doing something wrong?

Reply:
Didn't work for me at first, I thought it was fine now. Can't figure it out right now, it seems it might be a common problem.. https://stackoverflow.com/questions/13337599/function-onopen-is-not-running